.github/workflows/ci.yaml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  push:
+    branches: [ doc_gen ]
+  pull_request:
+    branches: [ doc_gen ]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            rosdistro: dashing
+          - os: ubuntu-18.04
+            rosdistro: eloquent
+          - os: ubuntu-20.04
+            rosdistro: foxy
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ros-tooling/setup-ros@0.0.23
+      with:
+        required-ros-distributions: ${{ matrix.rosdistro }}
+    - name: Install dependencies
+      run: |
+        sudo apt update -qq
+        sudo apt install -y curl doxygen python3-pip
+        pip3 install sphinx sphinx-rtd-theme sphinx-autodoc-typehints
+    - name: Build code and docs
+      run: ${GITHUB_WORKSPACE}/build_docs.sh -y -r ${{ matrix.rosdistro }}
```

Comment:
```suggestion
      export PATH="/home/runner/.local/bin:$PATH"
      run: ${GITHUB_WORKSPACE}/build_docs.sh -y -r ${{ matrix.rosdistro }}
```

Reply:
Thanks. I thought maybe using the `setup-python` action would handle this for us, but it seems to have introduced other problems (cc49727).

Applied you're suggestion in  3df9b60